### ISSUE TITLE
Mac: Cmd+` not quite right

### ISSUE THREAD

**Issue by JohnScholes** [Dyalog/ride#328]
### Describe the issue you are having
A Mac convention is that repeated Cmd+backtick keystrokes cycle around windows within an app. In Ride, Cmd+` just switches to the session window and stops.

### Did you connect to an already running interpreter or start the interpreter from RIDE?
started an interpreter

### How do you reproduce the issue?
    )ed a b c
Cmd+` should cycle:  b c session a b c session ...

### Paste the contents of Help → About (Shift+F1)
IDE:
  Version: 4.1.2988
  Platform: MacIntel
  Date: 2017-12-14 21:00:41 +0100
  Git commit: b50942479e4d884ad23be5659c1ef7fece907db5
  Preferences:{
    "autocompletion":"0",
    "blinkCursor":"0",
    "colourScheme":"JohnS",
    "colourSchemes":"[{\"name\":\"JohnS\",\"styles\":\"diam=B err=fg:#ff0000,bg:#ffffff,bgo:0.5 idm=bg:#ffffff,bgo:0.5,fg:0 kw=B lnum=bg:f,bgo:1 mod=bg:#e7e1e7,bgo:1,fg:#000000 mtch=bg:c,bgo:0.5 norm=bg:f,bgo:1 ns=fg:8 num=fg:8 quad=fg:8 srch=bg:#0000ff,bgo:0.5 str=fg:#000000 tc=bg:#ffffff,bgo:1,fg:0 zld=fg:8 vtt=bc:aaa\"}]",
    "editWins":"{\"width\":1104,\"height\":633,\"x\":336,\"y\":23,\"ox\":0,\"oy\":0}",
    "editWinsRememberPos":"1",
    "floatOnTop":"1",
    "floating":"1",
    "fold":"0",
    "ilf":"0",
    "indentOnOpen":"0",
    "kbdLocale":"en_US_Mac",
    "lbar":"0",
    "lineNums":"0",
    "pfkeys":"[\"\",\"\",\"\",\"\",\"\",\"wsdiff''<ER>\",\"\",\"\",\"\",\"\",\")reset<ER>)save<ER>\",\"\",\"\"]",
    "sqp":"0",
    "valueTips":"0",
    "zoom":"5"
  }

Interpreter:
  Version: 17.0.32332
  Platform: Mac-64
  Edition: Unicode/64
  Date: Feb 27 2018 at 14:14:32

**Comment by e9gille** [comment]
Just tested on the monaco branch and it works as described above.

**Comment by jayfoad** [comment]
Related to #224?

**Comment by JohnScholes** [comment]
Yes, it certainly looks likes the same issue. I would offer to try both issues to see if they go away with Monaco but I've uninstalled it because the pop-ups were doing my head in.

**Comment by e9gille** [comment]
I've now tested on the master branch 4.1.2988 and it works as expected.
With floating windows enabled it cycles between separate browser windows. This is the default behaviour in MacOS.
If you want to cycle between docked windows (not enabling floating windows feature) then you do need to configure the keyboard shortcut for TB.

@JohnScholes your configuration above suggests you use floating windows and yet it doesn't work on your machine. I'm not sure how to debug this. I've tested changing my kbdLocale to match yours and also changed the input source in the OS to US but it works whatever I do.

**Comment by e9gille** [comment]
Hold on, my physical keyboard is Swedish and I usually work with a GB input language in the OS settings and GB kbdLayout in RIDE. I type the backtick using the key that is located above the TAB key (to the left of `1`). It has always produced the expected character and I can use it to define the keyboard shortcut for TB.

If I switch to US input language in the OS, the backtick appears as the key left of `Z` when typing in the RIDE editor windows, but when I define keyboard shortcuts in RIDE I get the backtick if I hit the key left of `1` still.

In short, I'm lost. I can see you haven't configured any keyboard shortcuts and floating windows is enabled.  Cmd+` should cycle through the windows!

**Comment by e9gille** [comment]
I think I've figured it out. With always on top enabled it doesn't cycle but stops on the main window. Nothing I can fix, so closing.
